core/Azure.Mcp.Core/src/Areas/Server/Commands/Discovery/ConsolidatedToolDiscoveryStrategy.cs

```diff
@@ -0,0 +1,185 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Reflection;
+using System.Text.Json;
+using Azure.Mcp.Core.Areas.Server.Models;
+using Azure.Mcp.Core.Areas.Server.Options;
+using Azure.Mcp.Core.Commands;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace Azure.Mcp.Core.Areas.Server.Commands.Discovery;
+
+/// <summary>
+/// Discovery strategy that exposes command groups as MCP servers.
+/// This strategy converts Azure CLI command groups into MCP servers, allowing them to be accessed via the MCP protocol.
+/// </summary>
+/// <param name="commandFactory">The command factory used to access available command groups.</param>
+/// <param name="options">Options for configuring the service behavior.</param>
+/// <param name="logger">Logger instance for this discovery strategy.</param>
+public sealed class ConsolidatedToolDiscoveryStrategy(CommandFactory commandFactory, IOptions<ServiceStartOptions> options, ILogger<ConsolidatedToolDiscoveryStrategy> logger) : BaseDiscoveryStrategy(logger)
+{
+    private readonly CommandFactory _commandFactory = commandFactory;
+    private readonly IOptions<ServiceStartOptions> _options = options;
+
+    /// <summary>
+    /// Gets or sets the entry point to use for the command group servers.
+    /// This can be used to specify a custom entry point for the commands.
+    /// </summary>
+    public string? EntryPoint { get; set; } = null;
+    public static readonly string[] IgnoredCommandGroups = ["server", "tools"];
+
+    /// <summary>
+    /// Discovers available command groups and converts them to MCP server providers.
+    /// </summary>
+    /// <returns>A collection of command group server providers.</returns>
+    public override Task<IEnumerable<IMcpServerProvider>> DiscoverServersAsync()
+    {
+        // Load consolidated tools from JSON file
+        var consolidatedTools = new List<ConsolidatedToolDefinition>();
+        try
+        {
+            var assembly = Assembly.GetExecutingAssembly();
+            var resourceName = "Azure.Mcp.Core.Areas.Server.Resources.consolidated-tools.json";
+            using var stream = assembly.GetManifestResourceStream(resourceName);
+            if (stream != null)
+            {
+                using var reader = new StreamReader(stream);
+                var json = reader.ReadToEnd();
+                var jsonDoc = JsonDocument.Parse(json);
+                if (jsonDoc.RootElement.TryGetProperty("consolidated_tools", out var toolsArray))
+                {
+                    consolidatedTools = JsonSerializer.Deserialize(toolsArray.GetRawText(), ServerJsonContext.Default.ListConsolidatedToolDefinition) ?? new List<ConsolidatedToolDefinition>();
+                }
```

Comment:
Should we just error out here if the expected JSON property can't be found? Does anything good happen if this is missing and we continue to load the server?

Reply:
Added proper error handling around the json file parsing.